app/controllers/servers_controller.rb

```diff
@@ -23,114 +22,44 @@ def new
-        format.json { render :show, status: :created, location: @server }
-      else
-        format.html { render :new }
-        format.json { render json: @server.errors, status: :unprocessable_entity }
-      end
+    @server = Server.new(server_params)
+    if @server.save
+      redirect_to @server, notice: 'Server was successfully created.'
+    else
+      render :new
     end
   end
-
-  def setNewServerObject
-    @server = Server.new(
-      name: server_params[:name],
-      cpu_cores: server_params[:cpu_cores],
-      ram_gb: server_params[:ram_gb],
-      storage_gb: server_params[:storage_gb],
-      mac_address: server_params[:mac_address],
-      fqdn: server_params[:fqdn],
-      ipv4_address: server_params[:ipv4_address],
-      ipv6_address: server_params[:ipv6_address],
-      installed_software: server_params[:installed_software],
-      vendor: server_params[:vendor],
-      model: server_params[:model],
-      description: server_params[:description],
-      responsible_id: server_params[:responsible]
-    )
-  end
-
+  
   # PATCH/PUT /servers/1
-  # PATCH/PUT /servers/1.json
   def update
-    respond_to do |format|
-      if @server.update(server_params.permit(
-                          :name,
-                          :cpu_cores,
-                          :ram_gb,
-                          :storage_gb,
-                          :ipv4_address,
-                          :ipv6_address,
-                          :mac_address,
-                          :fqdn,
-                          :installed_software,
-                          :model,
-                          :vendor,
-                          :description,
-                          :responsible_id
-                        ))
-        format.html { redirect_to @server, notice: 'Server was successfully updated.' }
-        format.json { render :show, status: :ok, location: @server }
-      else
-        format.html { render :edit }
-        format.json { render json: @server.errors, status: :unprocessable_entity }
-      end
+    if @server.update(server_params)
+      redirect_to @server, notice: 'Server was successfully updated.'
+    else
+      render :edit
     end
   end
-
+  
   # DELETE /servers/1
-  # DELETE /servers/1.json
   def destroy
     @server.destroy
-    respond_to do |format|
-      format.html { redirect_to servers_path, notice: 'Server was successfully deleted.' }
-      format.json { head :no_content }
-    end
+    redirect_to servers_path, notice: 'Server was successfully deleted.'
   end
 
   private
-
+  
   # Use callbacks to share common setup or constraints between actions.
   def set_server
     @server = Server.find(params[:id])
   end
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def server_params
-    params.fetch(:server, {})
+    # Remove empty software form fields
+    params[:server][:installed_software].reject!(&:blank?) if params[:server][:installed_software]
```

Comment:
Won't this crash if params[:server] is nil?
We should to check for this in the condition

Reply:
> Won't this crash if `params[:server]` is nil?

Yes. If no `server` param is given and this line wasn't there, there would be an error one line later in the `require` call: https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-require

Reply:
But good point, the original exception is neater. I've dug around in `ActionController::Parameters` and found an acceptable way of changing params after permit.